### DISCUSSION TITLE
measuring audit api requests?

### DISCUSSION THREAD

**Discussion by thomasmckay** [open-policy-agent/community#209] [Gatekeeper]
```
Audit chunk size: set --audit-chunk-size=400 (defaults to 500, 0 = infinite)
Lower chunk size can reduce memory consumption of the auditing Pod 
but can increase the number requests to the Kubernetes API server.
```
from [docs](https://open-policy-agent.github.io/gatekeeper/website/docs/audit/#configuring-audit)

is there a metric to measure the requests? how would someone "dial in" an optimal chunk/mem vs. api?

**Comment by sozercan** [comment]
@thomasmckay afaik there is no metric in gatekeeper that measures API server requests for audit at this time. Supported metrics are available here: https://open-policy-agent.github.io/gatekeeper/website/docs/metrics

Kubernetes API Server has a metric called `kube_apiserver.apiserver_request_count` which might be useful to you.

There's no magic bullet for optimal chunk size vs API. It's a tradeoff where you want less memory consumption vs API requests.
